compiler+runtime/src/cpp/jank/runtime/obj/persistent_string_sequence.cpp

```diff
@@ -0,0 +1,113 @@
+#include <jank/runtime/obj/persistent_string_sequence.hpp>
+
+namespace jank::runtime
+{
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s)
+    : str{ s }
+  {
+    assert(s->count() > 0);
+  }
+
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s, size_t i)
+    : str{ s }
+    , index{ i }
+  {
+    assert(s->count() > 0);
+  }
+
+  /* behavior::objectable */
+  native_bool obj::persistent_string_sequence::equal(object const &o) const
+  {
+    if(o.type != object_type::persistent_string_sequence)
+    {
+      return false;
+    }
+    auto const seq(expect_object<obj::persistent_string_sequence>(&o));
+    return std::equal(str->data.begin() + index,
+                      str->data.end(),
+                      seq->str->data.begin() + seq->index,
+                      seq->str->data.end());
```

Comment:
Hmmm, I see. I think I've led you astray here, since I didn't quite see what you needed. In my example of `persistent_string::equal`, it will check if the rhs is a string first, since a string will only every be equal to a string. Sequences are different though. Check it:

```
user=> (= "foo" '(\f \o \o))
false
user=> (= (seq "foo") '(\f \o \o))
true
```

So, we need to `visit_seqable` on rhs and walk through the sequences, comparing by using your new `equal(char, object_ptr)` fn. There's a smart way we can do this, but it will require some more explanation.